src/ripple/server/impl/BaseWSPeer.h

```diff
@@ -195,7 +203,12 @@ run()
             &BaseWSPeer::run, impl().shared_from_this()));
     impl().ws_.set_option(beast::websocket::decorate(identity{}));
     impl().ws_.set_option(port().pmd_options);
+    impl().ws_.set_option(beast::websocket::ping_callback{
+        std::bind(&BaseWSPeer::on_ping_pong, this,
+            std::placeholders::_1, std::placeholders::_2)});
```

Comment:
Consider lambdas in place of `bind`:
````
[this](auto x, auto const& y) {this->on_ping_pong(x, y);}});
````
Technically the `this->` in the lambda body isn't needed, but gcc won't support that until gcc-7.0.  I tested the above on all three compilers and it appears to work fine.